src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
@serge-groshev `jedis.close()` automatically returns the `Jedis` instance to it's corresponding pool accordingly. If the Jedis instance is dirty (an error raised trying to use it) the `close` method will take care of returning the broken resource automatically. 

This way you don't have to deal with knowing which state the connection has before returning it to the pool as it's being done automatically.